src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Guid.txt

```diff
@@ -0,0 +1,40 @@
+﻿// Guid function 
+// https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-guid
+
+// Guid with no args is volatile, so we can't directly test the output
+
+>> IsBlank(GUID())
+false
+
+>> IsError(GUID())
+false
+
+// Multiple invocations are unique
+>> GUID() <> GUID()
+true
+
+// Coercion to string is 32 chars + 4 hyphens
+>> Len(GUID())
+36
+
+>> GUID("c3987aa4-52ff-462f-9a7c-a13c0dedb16d")
+c3987aa4-52ff-462f-9a7c-a13c0dedb16d
+
+>> GUID("8A9E8006-3EDE-4584-863F-4226BAA73D08")
+8a9e8006-3ede-4584-863f-4226baa73d08
+
+>> GUID("0A6D72E27FDE47C5BB9513BDB51BD7C0")
+0a6d72e2-7fde-47c5-bb95-13bdb51bd7c0
+
+>> GUID("0c99c9a834ed4247861dc9680d72dfd9")
+0c99c9a8-34ed-4247-861d-c9680d72dfd9
+
+>> Text(GUID("41aacfaa-8630-4d9b-8cdd-f62ed8ee8581"))
+"41aacfaa-8630-4d9b-8cdd-f62ed8ee8581"
+
+>> GUID("76353e29-1d57-4315-8af1-884ce7e6c75c") = GUID("76353e29-1d57-4315-8af1-884ce7e6c75c")
```

Comment:
>6353e29-1d57-4315-8af1-884ce7e6c75 [](http://example.com/codeflow?start=57&length=34)

Some negative tests (expecting error):

- GUID("not a guid")
- GUID("12345678-1234-1234-1234-12345678901") // missing one char
- GUID("12345678-1234-1234-1234-1234567890123") // one extra char
- GUID("12345678-gggg-1234-1234-123456789012") // characters other than 0-9a-fA-F